src/bucket/BucketManagerImpl.cpp

```diff
@@ -184,7 +184,14 @@ BucketManagerImpl::adoptFileAsBucket(std::string const& filename,
         {
             std::string err("Failed to rename bucket :");
             err += strerror(errno);
-            throw std::runtime_error(err);
+            // there is a race condition with external systems (windows)
+            // retry after sleeping for a second seems to work around the problem
+            std::this_thread::sleep_for(std::chrono::seconds(1));
+            if (rename(filename.c_str(), canonicalName.c_str()) != 0)
+            {
+                // if rename fails again, surface the original error
+                throw std::runtime_error(err);
```

Comment:
What message should the error contain? Should it be from the first failure, the second failure, or both? If it were to fail twice, I don't think it is necessarily true that both failures would be the same.

Reply:
I decided to use the original error as it's the reason we're failing in the first place